sotodlib/tod_ops/fft_ops.py

```diff
@@ -295,17 +300,38 @@ def calc_psd(
         merge_suffix (str, optional): Suffix to append to the Pxx field name in aman. Defaults to None (merged as Pxx).
         overwrite (bool): if true will overwrite f, Pxx axes.
         subscan (bool): if True, compute psd on subscans.
+        full_output: if True this also outputs nseg, the number of segments used for
+            welch, for correcting bias of median white noise estimation by calc_wn.
         **kwargs: keyword args to be passed to signal.welch().
 
     Returns:
         freqs: array of frequencies corresponding to PSD calculated from welch.
         Pxx: array of PSD values.
+        nseg: number of segments used for welch. this is returned if full_output is True.
     """
     if signal is None:
         signal = aman.signal
+
+    if ("noverlap" not in kwargs) or \
+            ("noverlap" in kwargs and kwargs["noverlap"] != 0):
+        logger.warning('calc_wn will be biased. noverlap argument of welch '
+                       'needs to be 0 to get unbiased median white noise estimate.')
+    if not full_output:
+        logger.warning('calc_wn will be biased. full_output argument of calc_psd '
+                       'needs to be True to get unbiased median white noise estimate.')
+
     if subscan:
-        freqs, Pxx = _calc_psd_subscan(aman, signal=signal, freq_spacing=freq_spacing, **kwargs)
+        if full_output:
+            freqs, Pxx, nseg = _calc_psd_subscan(aman, signal=signal,
+                                                 freq_spacing=freq_spacing,
+                                                 full_output=True,
+                                                 **kwargs)
+        else:
+            freqs, Pxx = _calc_psd_subscan(aman, signal=signal,
+                                           freq_spacing=freq_spacing,
+                                           **kwargs)
```

Comment:
does this need to be in an `if/else`? Can't you just pass `full_output=full_output`?

Reply:
This is because returned values are different.